File: tournaments/maxSubarray/maxSubarray.js
function maxSubarray(inputArray) {
    let result = 0;
    let a = 0;

    for (let i = 0; inputArray.length > i; ++i) {
        a = Math.max(0, a + inputArray[i]);
        result = Math.max(result, a);
    }

    return result;
}

File: tournaments/maxSubarray/maxSubarray.cpp
int maxSubarray(std::vector<int> inputArray) {
    int currentMax = 0;
    int mx = 0;

    for (int i = 0; i < inputArray.size(); i++) {
        currentMax = std::max(0, currentMax + inputArray[i]);
        mx = std::max(mx, currentMax);
    }

    return mx;
}
